src/test/ui/typeck/issue-24151.rs

```diff
@@ -0,0 +1,9 @@
+// check-pass
+
+fn main() {
+    unsafe {
+        let my_num: &mut [i32; 2] = &mut [10, 20];
+        let my_num: *mut i32 = my_num as *mut i32;
+        println!("{}", *my_num.offset(1));
```

Comment:
can you move this test to `ui/cast` instead and convert it to `run-pass`?
```suggestion
        assert_eq!(*my_num.offset(1), 20);
```

Reply:
Thanks for the suggestion. Done.